CanaryValidator/Canary.Tests.ps1

```diff
@@ -390,7 +390,85 @@ while ($runCount -le $NumberOfIterations)
+	
+	Invoke-Usecase -Name 'RoleAssignmentAndCustomRoleDefinition' -Description "Assign a reader role and create a custom role definition" -UsecaseBlock `
+        {
+            $readerRoleName = (Get-AzureRmRoleDefinition -Name Reader).Name
+            $subscriptionID = (Get-AzureRmSubscription -SubscriptionName $tenantSubscriptionName).SubscriptionId
+            $servicePrincipal = (Get-AzureRmADServicePrincipal)[0]
+            $customRoleName = "CustomCanaryRole-" + [Random]::new().Next(1,99)
+
+            Invoke-Usecase -Name 'ListAssignedRoles' -Description "List assigned roles to Service Principle - $($servicePrincipal.DisplayName)" -UsecaseBlock `
+            {
+	            Get-AzureRmRoleAssignment -ObjectId $servicePrincipal.Id -ErrorAction Stop
+            }
+
+            Invoke-Usecase -Name 'AssignReaderRole' -Description "Assign Reader role to Service Principle - $($servicePrincipal.DisplayName)" -UsecaseBlock `
+            {
+	            New-AzureRmRoleAssignment -Scope "/Subscriptions/$subscriptionID" -RoleDefinitionName $readerRoleName -ObjectId $servicePrincipal.Id -ErrorAction Stop
+                $assignedRoles = Get-AzureRmRoleAssignment -RoleDefinitionName $readerRoleName -Scope "/Subscriptions/$subscriptionID" -ErrorAction Stop
+                if (-not $assignedRoles)
+                {
+                    throw [System.Exception] "Unable to assign role ($readerRoleName) to Service Principle ($servicePrincipal.Id) for subscription $tenantSubscriptionName"
+                }                    
+            }
+
+            Invoke-Usecase -Name 'ListRoleDefinitions' -Description "List Role Definitions" -UsecaseBlock `
+            {
+	            Get-AzureRmRoleDefinition -ErrorAction Stop
+            }
+            
+            Invoke-Usecase -Name 'CustomRoleDefinition' -Description "Create a custom Role Definition - $customRoleName" -UsecaseBlock `
+            {
+	            $role = Get-AzureRmRoleDefinition -Name Reader                
+                $role.Id = $null                
+                $role.Name = $customRoleName
+                $role.Description = "Custom role definition for Canary"
+                $role.Actions.Clear()
+                $role.Actions.Add("Microsoft.Authorization/*/Read")
+                $role.AssignableScopes.Clear()
+                $role.AssignableScopes.Add("/Subscriptions/$subscriptionID")
+                New-AzureRmRoleDefinition -Role $role -ErrorAction Stop
+                $customRole = Get-AzureRmRoleDefinition -Name $customRoleName -Scope "/Subscriptions/$subscriptionID" -ErrorAction Stop
+                if (-not $customRole)
+                {
+                    throw [System.Exception] "Unable to create custom role ($customRoleName) for subscription $tenantSubscriptionName"
+                }    
+            }
 
+            Invoke-Usecase -Name 'GetProviderOperations' -Description "Get provider operations for all resource providers" -UsecaseBlock `
```

Comment:
This test in this form is not very useful. It must fail if any RP returns zero operation. You may want to add an exclusion list for now, but make sure it is removed once each RP adds its operation list.